docs/DEVELOPING.md

```diff
@@ -717,6 +717,188 @@ $ make ftest NAME=mssql
 
+
+## SharePoint Connector
+
+The [Elastic SharePoint connector](../connectors/sources/sharepoint.py) is provided in the Elastic connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
+
+### Availability and prerequisites
+
+⚠️ _Currently, this connector is available in **beta** starting in 8.8_.
+Features in beta are subject to change and are not covered by the service level agreement (SLA) of features that have reached general availability (GA).
+
+Elastic versions 8.6.0+ are compatible with Elastic connector frameworks. Your deployment must include the Elasticsearch, Kibana, and Enterprise Search services.
+
+### Compatibility
+
+  Sharepoint Connector supports below Sharepoint Server versions:
+  - Sharepoint 2013
+  - Sharepoint 2016
+  - Sharepoint 2019
+
+
+### Create an Administrator Account For SharePoint Online:
+  - To create an administrator account, join the Microsoft 365 Developer Program and get the Microsoft 365 developer subscription with a new administrator account for 90 days. Use this administrator account for signing into SharePoint Online and Azure AD to fetch more details.
+  - Register your app with Microsoft Azure AD
+  - To use the Microsoft identity platform endpoint, you must register your app using the Azure app registration portal. You can use either a Microsoft account or a work or school account to register an app.
+  - To configure an app to use the OAuth 2.0 authorization code grant flow, you'll need to save the following values when registering the app:
+    - Client Id: `Application(client) ID` which you can find inside App registrations from azure portal
+    - Client Secret: It is a `Value` which you need to create  form `Certificates and secrets` from azure portal
+    - Tenant Id: Dedicated instance of azure AD  which you can find inside App registrations from azure portal
+    - Tenant name: Name of domain
+
+
+### SharePoint Online permissions required to run the connector:
+
+   Refer to the following documentation for setting [SharePoint Permissions](https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs)
+
+   - To set `DisableCustomAppAuthentication` to false we need to connect to SharePoint using Windows PowerShell and then run set-spotenant -DisableCustomAppAuthentication $false
+
+   - In azure portal > Azure Active Directory > App registrations > API permission
+        - Click Add a permission
+        - Click Microsoft Graph and complete the following steps:
+            - Click Application permissions.
+            - Select the User.Read permission.
+        - After adding permission Grant admin consent for tenant_name    
+   - To assign full permissions to the tenant in SharePoint Online, in your browser, go to the tenant URL. For example, go to `https://<office_365_tenant_URL>/_layouts/15/appinv.aspx`. The SharePoint admin center page appears.
```

Comment:
I hit an issue where I kept getting an error that only a tenant admin could trust this app.

![Screenshot 2023-05-02 at 11 12 22 AM](https://user-images.githubusercontent.com/5288246/235727860-8b0bd38e-f873-4f54-8299-d422dfa82d9d.png)

I got around this following instructions from this [stack exchange post](https://sharepoint.stackexchange.com/a/280235/107970) which suggested using `https://[tenant]-admin.sharepoint.com/_layouts/15/appinv.aspx` (note the `-admin`) instead.